### ISSUE TITLE
Add support for Nvidia Shield Tablet

### ISSUE THREAD

**Issue by bluegizmo83** [sriharshaarangi/BatteryChargeLimit#28]
I posted in the forum, but I'm adding here also.

I found the correct file to make this work on the Nvidia Shield Tablet (Android version 7.0 stock rooted OS).

File location: /sys/devices/platform/7000c400.i2c/i2c-1/1-006b/charging_state
File has one line of text and only one word: "enabled" or "disabled"

Setting text to "enabled", which is the default text, allows charging, and setting it to "disabled" stops charging.

Took me almost two hours to find the correct file and test it.

**Comment by milux** [comment]
Please check out version 1.0.2 and report if it's working.

**Comment by bluegizmo83** [comment]
Testing now, I'll give a full report later today. So far though, upon first run after installing, it forced closed right away. It loaded up fine after that first crash though, and hasn't happened again yet. Also, the Reset Battery Stats button isn't working when SELinux is set to Enforcing. It works fine though if I set SELinux to Permissive (I had the same issue with the Automate Flow I created to control battery level and reset stats, so it's not specifically an issue with your app).

**Comment by milux** [comment]
Thanks for the logcat, this crash looks interesting...

**Comment by bluegizmo83** [comment]
Your welcome! One other thing I noticed while I was working on that Automate Flow for this tablet... On the Shield Tablet, when the charge cable is connected and the control file is set to "disabled", as soon as the charge cable is removed, the control file immediately reverts back to "enabled". This happens even with no Automate Flow or App enabled, just by manually editing the control file to "disabled" while the cable is connected, as soon as I remove the cable the control file reverts right back to "enabled". That doesn't happen on my Nexus 6p phone, so I'm not sure if it's just a weird thing Nvidia did, or if most devices do that. It doesn't seem to cause any issues or weird behavior so far though.

**Comment by milux** [comment]
Afaik, this happens for some devices and should not be a problem. When power supply is removed, the app does tries to do exactly that anyway.

**Comment by bluegizmo83** [comment]
Ok cool. Just thought I'd mention it. So far your app seems to be working as expected on this device, it's charging when it should, and stopping when it should. If I run into anymore issues or unexpected behavior I'll report back immediately, but it seems good to go.

**Comment by bluegizmo83** [comment]
Don't worry a whole lot about that one crash if it's too complicated or weird. I think I have some weird underlining issue on my tablet from a bad Titanium Backup restore... I recently upgrade from KitKat to Nougat on this tablet, and even though I did a full wipe between the two, I made the mistake of doing a full restore in TB of all user+system apps and data (and I started getting weird crashes occasionally after that). So that one crash with your app may not have even had anything to do with your app.
